src/vos/vos_obj.c

```diff
@@ -201,12 +283,17 @@ update_read_timestamps(struct vos_ts_set *ts_set, daos_epoch_t epoch,
 		entry->te_ts_rl = MAX(entry->te_ts_rl, epoch);
 		return;
 	}
+	if (entry->te_punch_propagated)
+		entry->te_ts_rl = MAX(entry->te_ts_rl, epoch);
```

Comment:
(style)  'te' may be misspelled - perhaps 'the'?
(style)  'te' may be misspelled - perhaps 'the'?